primitives/src/signatures/bls.rs

```diff
@@ -1,24 +1,277 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Jellyfish library.
+
+// You should have received a copy of the MIT License
+// along with the Jellyfish library. If not, see <https://mit-license.org/>.
+
 //! BLS Signature Scheme
+//!
+//! Conforming to [IRTF draft][irtf], wrapping [`blst` crate][blst] under the
+//! hood.
+//!
+//! [irtf]: https://datatracker.ietf.org/doc/pdf/draft-irtf-cfrg-bls-signature-05
+//! [blst]: https://github.com/supranational/blst
+//!
+//! # Examples
+//!
+//! ```
+//! use rand_core::{RngCore, OsRng};
+//! use jf_primitives::signatures::{SignatureScheme, bls::BLSSignatureScheme};
+//!
+//! let pp = BLSSignatureScheme::param_gen::<OsRng>(None)?;
+//!
+//! // make sure the PRNG passed in is securely seeded, we RECOMMEND using `OsRng`
```

Comment:
We should be careful about this recommendation. 

From the doc, it says
```
You may prefer to use [EntropyRng](https://docs.rs/rand/0.5.0/rand/rngs/struct.EntropyRng.html) instead of OsRng. It is unlikely, but not entirely theoretical, for OsRng to fail. In such cases [EntropyRng](https://docs.rs/rand/0.5.0/rand/rngs/struct.EntropyRng.html) falls back on a good alternative entropy source.
```

Reply:
Oh nvm that EntropyRng is deprecated in the newest version.
My point is, we should only make concrete recommendation if we are sure about it.

Reply:
The main reasons that `OsRng` might fail are: 1) process gets blocked since it was just initialized and entropy pool is still warming up; 2) in an environment that doesn't give you such access, such as a restricted wasm. 

On one hand, I'm fairly confident that this is a secure recommendation. 
OTOH, I can tone down a bit and say "you could use something like `OsRng` or seed your `SeedableRng` with some os-level randomness".

do you have suggestions on how should I rephrase it?

Reply:
>OTOH, I can tone down a bit and say "you could use something like OsRng or seed your SeedableRng with some os-level randomness".

I think this is a good one. Plus a warning that user should use a high entropy random seed in production.